src/util/prolog/prologResult.ts

```diff
@@ -9,22 +9,32 @@ const booleanAnswerRegExp = /^(true|false)( ;|,|.)$/;
 // Second group contains the variable names
 const variableNameRegExp = /(^| )([A-Z]\w*)/g;
 
+/** Handles the results for a prolog query returned by PrologSession.executeQuery() */
 export class PrologResult {
 	private readonly rawResults: string[];
 	private results: Map<string, Array<string | boolean>> | undefined;
 
+	/**
+		* Simply forwards the raw results to its respective attribute
+		* @param {Array<string>} rawResults - The string array containing the results
+		*/
 	constructor(rawResults: string[]) {
 		this.rawResults = rawResults;
 	}
 
-	public getRawResults() {
+	/**
+		* rawResults getter-Methode
+		* @return {Array<string>} The raw results
+		*/
+	public getRawResults(): string[] {
 		return this.rawResults;
 	}
 
 	/**
-		* Generates (if needed) and returns the results.
-		*
-		*
+		* Creates and returns curated results
```

Comment:
What do you mean by "curated results"? (Note: Maybe my english vocabulary isn't good enough)

Reply:
I guess it is not a frequently used word? It means the results are not just anything raw but they now more selective and prettied up.

Usually you curate an exhibition.

Reply:
Ah, so in my lingo I would declare them as "filtered" or something like that. But I guess your description fits at least as well.